applications/launchpad/backend/src/grpc/mod.rs

```diff
@@ -61,13 +64,21 @@ impl TryFrom<TransactionEvent> for WalletTransaction {
     }
 }
 
-impl From<GetIdentityResponse> for WalletIdentity {
-    fn from(value: GetIdentityResponse) -> WalletIdentity {
-        WalletIdentity {
+impl TryFrom<GetIdentityResponse> for WalletIdentity {
+    type Error = String;
+
+    fn try_from(value: GetIdentityResponse) -> Result<Self, Self::Error> {
+        let hex_public_key = String::from_utf8(value.public_key.clone()).unwrap();
+        let emoji_id = EmojiId::from_hex(&hex_public_key)
+            .map_err(|e| format!("Failed to create an emoji: {}", e))?
+            .as_str()
+            .to_string();
```

Comment:
You don't need the as_str() call

```suggestion
            .to_string();
```